app/src/main/java/com/example/vmedvediev/redditapp/comments/CommentsActivity.kt

```diff
@@ -191,25 +189,17 @@ class CommentsActivity : AppCompatActivity() {
             headerMap["User-Agent"] = username
             headerMap["X-Modhash"] = modhash
             headerMap["cookie"] = "reddit_session=$cookie"
-        
-            val call = initRetrofit(GsonConverterFactory.create()).submitComment(headerMap, "comment", postId, comment)
-            call.enqueue(object : Callback<CommentChecker> {
-                override fun onResponse(call: Call<CommentChecker>?, response: Response<CommentChecker>?) {
-                    Log.e(TAG, "onResponse: SERVER RESPONSE: ${response.toString()}")
-                    val success = response?.body()?.success
-
-                    if (success == "true") {
-                        handleSuccessPostComment()
-                    } else {
-                        handleUnsuccessPostComment()
-                    }
-                }
 
-                override fun onFailure(call: Call<CommentChecker>?, t: Throwable?) {
-                    Log.e(CommentsActivity.TAG, "onFailure: Unable to post comment: ${t?.message}")
-                    Toast.makeText(this@CommentsActivity, "An Error Occured!", Toast.LENGTH_SHORT).show()
-                }
-            })
+        launch(UI) {
+            try {
+                val success = initRetrofit(GsonConverterFactory.create())
+                        .submitComment(headerMap, "comment", postId, comment).await().success.toBoolean()
+                return@launch if (success) handleSuccessPostComment() else handleUnsuccessPostComment()
```

Comment:
Just an example of code structure
`// dispatches execution onto the Android main UI thread
val uiContext: CoroutineContext = UI

// represents a common pool of shared threads as the coroutine dispatcher
val bgContext: CoroutineContext = CommonPool
`

`fun fetchMyContacts() = launch(uiContext) {
        val contactsList: MutableList<ContactModel> = ArrayList()
        try {
            async(bgContext) {
                fetchContactsList(contactsList)
               }.await()
            myContactsView.setMyContactsList(sortedContactsList)
          } catch (exception: Exception) {
            Timber.e(exception, "fetchMyContacts: ")
            myContactsView.setMyContactsList(sortedContactsList)
        }
    }`